<Type Name="XamlInternalException" FullName="System.Xaml.XamlInternalException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="97d8995fcf52713d0383a0f9af67e24a7dc31af6" /><Meta Name="ms.sourcegitcommit" Value="1b924db57b3a1cf768d98c21f9b988d6966a0f2b" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="ja-JP" /><Meta Name="ms.lasthandoff" Value="03/06/2020" /><Meta Name="ms.locfileid" Value="78732115" /></Metadata><TypeSignature Language="C#" Value="public class XamlInternalException : System.Xaml.XamlException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit XamlInternalException extends System.Xaml.XamlException" FrameworkAlternate="netcore-3.0;netcore-3.1" />
  <TypeSignature Language="DocId" Value="T:System.Xaml.XamlInternalException" />
  <TypeSignature Language="VB.NET" Value="Public Class XamlInternalException&#xA;Inherits XamlException" />
  <TypeSignature Language="C++ CLI" Value="public ref class XamlInternalException : System::Xaml::XamlException" />
  <TypeSignature Language="F#" Value="type XamlInternalException = class&#xA;    inherit XamlException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit XamlInternalException extends System.Xaml.XamlException" FrameworkAlternate="netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8" />
  <AssemblyInfo>
    <AssemblyName>System.Xaml</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Xaml.XamlException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;netframework-4.8">
      <AttributeName>System.Serializable</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="075f8-101">XAML の読み取り中および XAML の書き込み中に発生する内部矛盾に対してスローされる例外です。</span><span class="sxs-lookup"><span data-stu-id="075f8-101">The exception that is thrown for internal inconsistencies that occur during XAML reading and XAML writing.</span></span></summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="075f8-102">この例外がスローされるのは、通常、XAML リーダーまたは XAML ライターシステムの一部がコードによって拡張され、.NET Framework XAML サービスのシステムコードに渡された不整合または実装の問題が発生した場合のみです。</span><span class="sxs-lookup"><span data-stu-id="075f8-102">This exception is usually thrown only when some aspect of the XAML reader or XAML writer system was extended by your code, and inconsistencies or implementation problems occurred that were passed on to the .NET Framework XAML Services system code.</span></span> <span data-ttu-id="075f8-103">XAML の解析、XAML でのエラーの発生など、サードパーティ製の例外は <xref:System.Xaml.XamlInternalException>をスローしないようにし、代わりに <xref:System.Xaml.XamlParseException> や <xref:System.Xaml.XamlObjectWriterException>などの他の例外をスローする必要があります。</span><span class="sxs-lookup"><span data-stu-id="075f8-103">Exceptions that might originate from third parties, such as parsing XAML and encountering errors in the XAML, should not throw <xref:System.Xaml.XamlInternalException>, and should instead throw other exceptions such as <xref:System.Xaml.XamlParseException> or <xref:System.Xaml.XamlObjectWriterException>.</span></span>  
  
 <span data-ttu-id="075f8-104">Xaml リーダーまたは XAML ライターの実装のコードを記述する場合は .NET Framework の XAML サービスを拡張するコードを記述する場合は、<xref:System.Xaml.XamlInternalException> とその他の XAML 関連の例外との違いを維持する必要があります。</span><span class="sxs-lookup"><span data-stu-id="075f8-104">If you are writing code that extends .NET Framework XAML Services, in particular, if you are writing code for a XAML reader or XAML writer implementation, you should maintain this distinction between <xref:System.Xaml.XamlInternalException> and the other XAML-related exceptions.</span></span> <span data-ttu-id="075f8-105">コードによってスローされるのは、クラスの拡張が呼び出し可能コードで定義されている設計上の原則や実装パターンに違反している場合だけです。 <xref:System.Xaml.XamlInternalException></span><span class="sxs-lookup"><span data-stu-id="075f8-105">Your code should only throw <xref:System.Xaml.XamlInternalException> in cases where further extension of your class violates some design principle or implementation pattern that your callable code defines.</span></span> <span data-ttu-id="075f8-106">コードでは、XAML リーダーまたは XAML ライターの実装が拡張なしで使用され、サードパーティの XAML、サードパーティの XAML スキーマコンテキストなどを読み込むために使用される例外の場合に、XAML 固有のその他の例外をスローする必要があります。</span><span class="sxs-lookup"><span data-stu-id="075f8-106">Your code should throw other XAML-specific exceptions for exception cases where your XAML reader or XAML writer implementation is used without extension and is used to load third-party XAML, third-party XAML schema context, and so on.</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Xaml.XamlParseException" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Xaml</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="075f8-107"><see cref="T:System.Xaml.XamlInternalException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="075f8-107">Initializes a new instance of the <see cref="T:System.Xaml.XamlInternalException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public XamlInternalException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Xaml.XamlInternalException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; XamlInternalException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Xaml</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="075f8-108">エラーを説明するシステム提供のメッセージを使用して、<see cref="T:System.Xaml.XamlInternalException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="075f8-108">Initializes a new instance of the <see cref="T:System.Xaml.XamlInternalException" /> class with a system-supplied message that describes the error.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public XamlInternalException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Xaml.XamlInternalException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; XamlInternalException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Xaml.XamlInternalException : string -&gt; System.Xaml.XamlInternalException" Usage="new System.Xaml.XamlInternalException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Xaml</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="075f8-109">例外を説明するメッセージ。</span><span class="sxs-lookup"><span data-stu-id="075f8-109">The message that describes the exception.</span></span></param>
        <summary><span data-ttu-id="075f8-110">エラーを説明する指定したメッセージを使用して、<see cref="T:System.Xaml.XamlInternalException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="075f8-110">Initializes a new instance of the <see cref="T:System.Xaml.XamlInternalException" /> class with a specified message that describes the error.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected XamlInternalException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; XamlInternalException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="new System.Xaml.XamlInternalException : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; System.Xaml.XamlInternalException" Usage="new System.Xaml.XamlInternalException (info, context)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Xaml</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="075f8-111">シリアル化されたオブジェクト データを保持するオブジェクト。</span><span class="sxs-lookup"><span data-stu-id="075f8-111">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="075f8-112">転送元または転送先に関するコンテキスト情報。</span><span class="sxs-lookup"><span data-stu-id="075f8-112">The contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="075f8-113">シリアル化したデータを使用して、<see cref="T:System.Xaml.XamlInternalException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="075f8-113">Initializes a new instance of the <see cref="T:System.Xaml.XamlInternalException" /> class with serialized data.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public XamlInternalException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; XamlInternalException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Xaml.XamlInternalException : string * Exception -&gt; System.Xaml.XamlInternalException" Usage="new System.Xaml.XamlInternalException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Xaml</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="075f8-114">例外を説明するメッセージ。</span><span class="sxs-lookup"><span data-stu-id="075f8-114">The message that describes the exception.</span></span></param>
        <param name="innerException"><span data-ttu-id="075f8-115">現在の例外の原因となった例外。</span><span class="sxs-lookup"><span data-stu-id="075f8-115">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="075f8-116"><paramref name="innerException" /> パラメーターが <see langword="null" /> でない場合は、内部例外を処理する <see langword="catch" /> ブロックで現在の例外が発生します。</span><span class="sxs-lookup"><span data-stu-id="075f8-116">If the <paramref name="innerException" /> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="075f8-117">指定したエラー メッセージおよびこの例外の原因となった内部例外への参照を使用して、<see cref="T:System.Xaml.XamlInternalException" /> クラスの新しいインスタンスを初期化します。</span><span class="sxs-lookup"><span data-stu-id="075f8-117">Initializes a new instance of the <see cref="T:System.Xaml.XamlInternalException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
  </Members>
</Type>
